src/plugins/announce.lua

```diff
@@ -0,0 +1,53 @@
+-- URL announcements - follows http->https redirects but not www
+
+local https = require 'ssl.https'
+local ltn12 = require 'ltn12'
+
+local plugin = {}
+
+plugin.help = 'Usage: announce <url>'
+
+plugin.main = function(args)
+    local _, _, url = args.message:find('(.+)')
+
+    if args.modules.config.debug then
+        print(url)
+    end
+
+    if not url then
+        args.modules.irc.privmsg(args.target, ('%s: Please give me a url.'):format(args.sender))
+        return
+    end
+
+    if url:find('[ <]?https?://[^> ]+.*') then
```

Comment:
I think this pattern should be changed to have `.-` in front of the `[ <]` class:

```lua
     if url:find('.-[ <]?https?://[^> ]+.*') then
```

This allows for handling the case where someone says a sentence that contains a URL.

Reply:
done!